internal/service/opensearchserverless/security_policy_data_source.go

```diff
@@ -0,0 +1,72 @@
+package opensearchserverless
+
+import (
+	"context"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-aws/internal/conns"
+	"github.com/hashicorp/terraform-provider-aws/internal/errs/sdkdiag"
+)
+
+// @SDKDataSource("aws_opensearchserverless_security_policy")
+func DataSourceSecurityPolicy() *schema.Resource {
+	return &schema.Resource{
+		ReadWithoutTimeout: dataSourceSecurityPolicyRead,
+
+		Schema: map[string]*schema.Schema{
+			"description": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"name": {
```

Comment:
The [API documentation](https://docs.aws.amazon.com/opensearch-service/latest/ServerlessAPIReference/API_GetSecurityPolicy.html#API_GetSecurityPolicy_RequestSyntax) has stated `length` and `patterns` can you please include those here on schema validation?

You can see an example here: https://github.com/hashicorp/terraform-provider-aws/blob/8f461fec79149dc1031ee843250ead0da9030483/internal/service/opensearchserverless/collection.go#L104C8-L104C8